### PR TITLE
chore: Add recommended VS Code extensions

### PR DESCRIPTION
**Pull request by nivisi**
This is the initial list of recommended VS Code extensions. Feel free to recommend others!

### CODE REVIEW

File: .vscode/extensions.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+        "dart-code.dart-code",
+        "dart-code.flutter",
+        "usernamehw.errorlens",
+        "felixangelov.mason",
+        "felixangelov.bloc",
+        "nivisi.dart-build-runner-tools",
+        "tenraneko.pubspec-dependency-opener",
+        "jeroen-meijer.pubspec-assist",
+    ]
+}
```
Comment (utpal-barman) [inline .vscode/extensions.json:12]: Enter a new line at the end

**Review by utpal-barman — APPROVED** [review]
Good idea! 👏

File: .vscode/extensions.json (hunk 1 above)
Comment (nivisi) [reply .vscode/extensions.json:13]: @utpal-barman why? My formatter automatically removes it

Comment (utpal-barman) [reply .vscode/extensions.json:13]: I guess it's POSIX where it detects each line when it ends with a new line character, GitHub has this feature enabled by default and suggests ending a new line on PR, Even Dart-formatter adds EOF by default. I don't like that red sign warning TBH 😄 

(Not sure why your IDE removes it, for me JSON file doesn't get formatted on a fresh IDE, maybe there are some plugins working on your side, if you want this feature by default you can install Prettier)

Screenshot 2024-05-14 at 3 59 01 PM

Comment (nivisi) [reply .vscode/extensions.json]: Done boss. Had to run the command explicitly though  😄

**Pull request merged**